webapp/src/websocket.ts

```diff
@@ -35,22 +64,37 @@ export default class WebSocketClient extends EventEmitter {
                 msg = JSON.parse(data);
             } catch (err) {
                 logErr('ws msg parse error', err);
+                return;
+            }
+
+            if (msg) {
+                this.serverSeqNo = msg.seq + 1;
             }
 
             if (!msg || !msg.event || !msg.data) {
                 return;
             }
 
             if (msg.event === 'hello') {
-                this.connID = msg.data.connection_id;
-                this.emit('open');
+                if (msg.data.connection_id !== this.connID) {
+                    logDebug('ws: new conn id from server');
+                    this.connID = msg.data.connection_id;
+                    this.serverSeqNo = 0;
+                    if (this.originalConnID === '') {
+                        logDebug('ws: setting original conn id');
+                        this.originalConnID = this.connID;
+                    }
+                }
+                if (!isReconnect) {
+                    this.emit('open', this.originalConnID, this.connID, false);
+                }
                 return;
             } else if (!this.connID) {
                 logWarn('ws message received while waiting for hello');
                 return;
             }
 
-            if (msg.data.connID !== this.connID) {
+            if (msg.data.connID !== this.connID && msg.data.connID !== this.originalConnID) {
```

Comment:
If I understand correctly, this means we're allowing the case where there are two valid connection ids for this client?

Reply:
This should really go away soon as we now support a way to send per connection ws events but for now, depending on which node a client is connected to and from where messages are coming from it may be either the original connection ID or the current one.